drivers/arm/gic/v2/gicv2_main.c

```diff
@@ -593,3 +593,11 @@ unsigned int gicv2_set_pmr(unsigned int mask)
 
 	return old_mask;
 }
+
+/*******************************************************************************
+ * This functions configures the single interrupt to be edge triggered
+ ******************************************************************************/
+void gicv2_interrupt_set_edge_triggered(unsigned int id)
```

Comment:
This seems to be useful during init-time only. In that case, the correct way to do this would be interrupt properties--edge/level trigger is one of the interrupt properties passed to various GIC init. See include/common/interrupt_props.h. The old method of GIC driver initialization is deprecated, and will be removed soon.

Reply:
Hi Marcin,
> It must be set each time we want to trigger another SPI via dedicated SoC control block. 

Okay, it seems you are triggering a non-secure interrupt from EL3. The expectation is that the default setup for NS interrupts done by firmware will be overwritten by the OS.  

Be aware that if the interrupt is pending when the property is changed, then the GIC specification says that the behaviour is undefined. Hence the recommendation to set this up one time.

>Otherwise, one-time config will be overriden e.g. during OS boot.

Ideally the OS should do the config for the NS interrupts. Is there some reason why the OS is not doing this correctly ?

> And I don't see any infrastructure to override this setting (if one-time config was the case).

Yes, there is no infrastructure to override the defaults setup for non secure interrupts due to the expectation I mention above.

> OK, I will update this PR once we agree on all code changes requested by reviewers.

Okay.

Best Regards
Soby Mathew

Reply:
Hi @soby-mathew 

I found out, that for PMU IRQ, the ACPI_MADT_PERFORMANCE_IRQ_MODE flag should enable setting edge trigger in OS. This will allow to remove calling gicv2_interrupt_set_edge_triggered() from the IRQ handler. Just need to verify it.

I hope the above works, but still I'd need to set the edge trigger for this IRQ one-time during init (in case UEFI wants to use this IRQ). In such case, would you accept using gicv2_interrupt_set_edge_triggered() in its current form or you prefer to override initial setting differently?

Thanks,
Marcin

Reply:
@soby-mathew @jeenu-arm 

Modifying ACPI MADT table succeeded and now we need only one time configuration of the SPI to be edge triggered. Please confirm if you are fine with:
  - removing gicv2_interrupt_set_edge_triggered() from the handler
  - calling one time during init in mvebu_pmu_interrupt_enable() a following:
```
       /* Configure ODMI Frame IRQs as edge triggered */
       for (idx = 0; idx < PLATFORM_CORE_COUNT; idx++)
               gicv2_interrupt_set_edge_triggered(A7K8K_ODMI_PMU_GIC_IRQ(idx));
```
If you don't have objections, Konstantin will submit a new version of the patchset.

Reply:
@semihalf-wojtas-marcin we think that it's better if the API is named gicv2_interrupt_set_cfg(), and take the interrupt number and configuration (edge/level).